src/solver/expressions/include/antares/solver/expressions/visitors/NodeVisitor.h

```diff
@@ -39,59 +40,71 @@ std::optional<RetT> tryVisit(const Node& node, VisitorT& visitor, Args... args)
+    {
+    }
 };
 
 template<class R, class... Args>
-class NodeVisitor
+class NodeVisitor: public IName
 {
 public:
     virtual ~NodeVisitor() = default;
 
-    R dispatch(const Node& node, Args... args)
+    R dispatch(const Nodes::Node& node, Args... args)
     {
-        using FunctionT = std::optional<R> (*)(const Node&, NodeVisitor<R, Args...>&, Args... args);
+        using FunctionT = std::optional<R> (*)(const Nodes::Node&,
+                                               NodeVisitor<R, Args...>&,
+                                               Args... args);
         static const std::vector<FunctionT> allNodeVisitList{
-          &tryVisit<R, NodeVisitor<R, Args...>, AddNode>,
-          &tryVisit<R, NodeVisitor<R, Args...>, SubtractionNode>,
-          &tryVisit<R, NodeVisitor<R, Args...>, MultiplicationNode>,
-          &tryVisit<R, NodeVisitor<R, Args...>, DivisionNode>,
-          &tryVisit<R, NodeVisitor<R, Args...>, EqualNode>,
-          &tryVisit<R, NodeVisitor<R, Args...>, LessThanOrEqualNode>,
-          &tryVisit<R, NodeVisitor<R, Args...>, GreaterThanOrEqualNode>,
-          &tryVisit<R, NodeVisitor<R, Args...>, NegationNode>,
-          &tryVisit<R, NodeVisitor<R, Args...>, ParameterNode>,
-          &tryVisit<R, NodeVisitor<R, Args...>, VariableNode>,
-          &tryVisit<R, NodeVisitor<R, Args...>, LiteralNode>,
-          &tryVisit<R, NodeVisitor<R, Args...>, PortFieldNode>,
-          &tryVisit<R, NodeVisitor<R, Args...>, ComponentVariableNode>,
-          &tryVisit<R, NodeVisitor<R, Args...>, ComponentParameterNode>};
+          &tryVisit<R, NodeVisitor<R, Args...>, Nodes::AddNode>,
+          &tryVisit<R, NodeVisitor<R, Args...>, Nodes::SubtractionNode>,
+          &tryVisit<R, NodeVisitor<R, Args...>, Nodes::MultiplicationNode>,
+          &tryVisit<R, NodeVisitor<R, Args...>, Nodes::DivisionNode>,
+          &tryVisit<R, NodeVisitor<R, Args...>, Nodes::EqualNode>,
+          &tryVisit<R, NodeVisitor<R, Args...>, Nodes::LessThanOrEqualNode>,
+          &tryVisit<R, NodeVisitor<R, Args...>, Nodes::GreaterThanOrEqualNode>,
+          &tryVisit<R, NodeVisitor<R, Args...>, Nodes::NegationNode>,
+          &tryVisit<R, NodeVisitor<R, Args...>, Nodes::ParameterNode>,
+          &tryVisit<R, NodeVisitor<R, Args...>, Nodes::VariableNode>,
+          &tryVisit<R, NodeVisitor<R, Args...>, Nodes::LiteralNode>,
+          &tryVisit<R, NodeVisitor<R, Args...>, Nodes::PortFieldNode>,
+          &tryVisit<R, NodeVisitor<R, Args...>, Nodes::ComponentVariableNode>,
+          &tryVisit<R, NodeVisitor<R, Args...>, Nodes::ComponentParameterNode>};
         for (auto f: allNodeVisitList)
         {
             if (auto ret = f(node, *this, args...); ret.has_value())
             {
                 return ret.value();
             }
         }
-        throw NodeVistorException("Antares::Solver::Nodes Visitor: invalid Node!");
+        throw InvalidNode("Antares::Solver::Nodes Visitor: invalid Node!");
```

Comment:
```cpp
throw InvalidNode("Antares::Solver::Nodes Visitor: invalid node type " + node.name());
```

Reply:
well, we"re in trouble if input Node& node is obtained from nullptr

Reply:
yes